avogadro/rendering/solidpipeline.cpp

```diff
@@ -157,12 +160,37 @@ void SolidPipeline::end()
                  d->depthTexture, m_width, m_height);
   d->firstStageShaders.setUniformValue("inAoEnabled", m_aoEnabled ? 1.0f : 0.0f);
   d->firstStageShaders.setUniformValue("inAoStrength", m_aoStrength);
+  d->firstStageShaders.setUniformValue("inFogStrength", (m_fogStrength));
   d->firstStageShaders.setUniformValue("inEdStrength", m_edStrength);
+  d->firstStageShaders.setUniformValue("inFogEnabled", m_fogEnabled ? 1.0f : 0.0f);
+  d->firstStageShaders.setUniformValue("inFogPosition", m_fogPosition);
+  d->firstStageShaders.setUniformValue("fogR", (m_backgroundColor[0])/255.0f);
+  d->firstStageShaders.setUniformValue("fogG", (m_backgroundColor[1])/255.0f);
+  d->firstStageShaders.setUniformValue("fogB", (m_backgroundColor[2])/255.0f);
   glDrawArrays(GL_TRIANGLES, 0, 6);
-
+  // std::cout<<m_fogStrength<<std::endl;
   glDisableVertexAttribArray(0);
 }
 
+// TODO: More precise calculations needed
+void SolidPipeline::adjustOffset(const Camera& cam) {
+
+  //since it's a workaround, I feel like I can optimize it more. 
```

Comment:
At the very least, it would be really helpful to know where / how these numbers came about? Otherwise they are just a bunch of "magic numbers" with no indication of the derivation and purpose. Can you please add some comments?